src/samples/CardReader/Program.cs

```diff
@@ -28,13 +30,34 @@ private static async Task Main()
                 .AddFilter("OSDP.Net", LogLevel.Debug);
         });
 
-        var deviceConfiguration = new DeviceConfiguration();
+        var deviceConfiguration = new DeviceConfiguration
+        {
+            Address = deviceAddress,
+            DefaultSecurityKeyAllowed = defaultSecurityKeyAllowed,
+            SecurityKey = securityKey
+        };
         
         // Replace commented out code for test reader to listen on TCP port rather than serial
         //var communications = new TcpOsdpServer(5000, baudRate, loggerFactory);
         var communications = new SerialPortOsdpServer(portName, baudRate, loggerFactory);
 
         using var device = new MySampleDevice(deviceConfiguration, loggerFactory);
+        device.DeviceComSetUpdated += async (sender, args) =>
+        {
+            Console.WriteLine("A command has been processed to update the communication settings.");
+            Console.WriteLine($"Old settings - Baud Rate {args.OldBaudRate} - Address {args.OldAddress}");
+            Console.WriteLine($"New settings - Baud Rate {args.NewBaudRate} - Address {args.NewAddress}");
+            Console.WriteLine("A command has been processed to update the communication settings.");
+
+            if (sender is MySampleDevice mySampleDevice && args.OldBaudRate != args.NewBaudRate)
```

Comment:
does this sample handle the case where device address is changed?  

originally I was envisioning that when there's a new com set (which requires the parent of the device to take action), the parent would just recreate the whole device instance (rather than stop/start listening) and pass down the new configuration.

This way is also okay... I think but, now it needs to consider that device needs to be responding to a different address after ComSet returns.

Reply:
Once we go to having multiple devices on one connection, we will likely need a more elegant solution.